docs/osf-trolling-list/build_process.md

```diff
@@ -0,0 +1,127 @@
+   hard to find which version of SDK works with version of coreboot tree.
+2. Validation of coreboot-sdk is not sufficient - for example recently python
+   fixes were merged [[2]](https://review.coreboot.org/c/coreboot/+/45265), somehow this change passed all QA checks, but SeaBIOS
+   use python (not python2 or python3) and all builds using this version of SDK
+   that compile SeaBIOS fail.
+3. coreboot-sdk is based on moving target Debian sid, it is close to impossible
+   to build the same Docker images at 2 different points in time.
+4. coreboot-sdk enforce given version of ACPI spec - this may not always be
+   good thing to use most recent compiler and update code accordingly, since it
+   can easily break OSes.
+
+### Why we care?
+
+1. Whenever we deliver code or service to community or customer we provide
+   build environment which, for the sake of quality user support, should be
+   stable.
+2. CI pipelines rely on those containers. Replacing docker image in CI pipeline
+   whenever something change defeat the purpose of having automation and
+   increase maintenance cost significantly and increase maintenance cost
+   significantly
+
+## Requirements for OSF dev and build process
+
+Open Source Firmware development and build environment SHOULD have:
+* meaningful release process, we describe our approach to OSF-related release
+  processes [here](#TBD)
+* meaningful validation process, at least basic build system and dev env
+  capabilities should be tested
+* meaningful revisions with clear explanation what software stack can be build
+  with what version of dev and build system, description should include
+  side-spec compliance e.g. ACPI, SMBIOS, UEFI etc.
+* reproducible process that works across systems and in long run
+* signature, so anyone can identify where build and dev env coming from and if
+  it is trustworthy
+
+## Ideas for solving above issues
+
+1. Docker images are not reproducible - this is known fact and if we are
+   extremely serious about stability, security and quality we should stop using
+   not reproducible build environment. Of course there is some work making
+   Docker images reproducible, but it opens Pandora's box
+   [[3]](https://elinux.org/images/6/62/Building-Container-Images-with-OpenEmbedded-and-the-Yocto-Project-Scott-Murray-Konsulko-Group-1.pdf)
+   of Yocto or Buildroot. Other path could be Nix or Guix, what is currently
+   pursued in community [OSFW #guix-buildstack channel](http://osfw.slack.com/)
+   - this is private channel, feel free to ping anyone from 3mdeb Team to join.
+2. coreboot toolchain is build by its own build system, which would be hard to
+   couple with Dockerfile or something else.
+3. Docker images can be signed using `docker trust` commands as described
+   [here](https://docs.docker.com/engine/security/trust/#signing-images-with-docker-content-trust).
+
+I'm not tracking all activity in community, so some claims may be plain
```

Comment:
Aside from a couple of notes regarding the Docker, looks good to me.

Reply:
Sure. Thanks for the review.